File: practice/src/main/java/com/decade/practice/controllers/advices/ValidationControllerAdvice.java
package com.decade.practice.controllers.advices;

import com.decade.practice.controllers.validation.ChatIdentifierValidator;
import com.decade.practice.model.domain.embeddable.ChatIdentifier;
import org.springframework.web.bind.WebDataBinder;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.InitBinder;

@ControllerAdvice
public class ValidationControllerAdvice {

      @InitBinder
      protected void initBinder(WebDataBinder binder) {
            if (binder.getTarget() instanceof ChatIdentifier) {
                  binder.addValidators(new ChatIdentifierValidator());
            }
      }
}

File: practice/src/main/java/com/decade/practice/controllers/advices/ValidationControllerAdvice.kt
package com.decade.practice.controllers.advices

import com.decade.practice.controllers.validation.ChatIdentifierValidator
import com.decade.practice.model.domain.embeddable.ChatIdentifier
import org.springframework.web.bind.WebDataBinder
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.InitBinder


@ControllerAdvice
class ValidationControllerAdvice {
      @InitBinder
      protected fun initBinder(binder: WebDataBinder) {
            if (binder.target is ChatIdentifier)
                  binder.addValidators(ChatIdentifierValidator())
      }
}
